proguard/src/main/java/com/flurry/proguard/UploadProguardMapping.java

```diff
@@ -0,0 +1,216 @@
+        String retval = checkUploadStatus(projectId, uploadId, token);
+        if (retval != null) {
+            throw new IOException("Failed to upload with Failure Reason: " + retval);
+        }
+        System.out.println("SUCCESS!!");
+    }
+
+    private File tarZipFile(File file) throws IOException {
+        File tarZippedFile = File.createTempFile("tar-zipped-file", ".tar.gz");
+        TarArchiveOutputStream taos = new TarArchiveOutputStream(new GZIPOutputStream(new BufferedOutputStream(
+                new FileOutputStream(tarZippedFile))));
+        putArchives(taos, file, "");
+        taos.finish();
+        taos.close();
+        return tarZippedFile;
+    }
+
+    private void putArchives(TarArchiveOutputStream taos, File file, String base) throws IOException {
+        if (file.isDirectory()) {
+            for (File f : file.listFiles()) {
+                putArchives(taos, f, base + file.getName() + "/");
+            }
+        } else {
+            // file is just a single file
+            taos.putArchiveEntry(new TarArchiveEntry(file, base + file.getName()));
+            IOUtils.copy(new FileInputStream(file), taos);
+            taos.closeArchiveEntry();
+        }
+    }
+
+    private String lookUpProjectId(String apiKey, String token) throws IOException {
+        String queryUrl = METADATA_BASE + "project?filter[project.apiKey]=" + apiKey + "&fields[project]=apiKey";
+        HttpGet getRequest = new HttpGet(queryUrl);
+        Map<String, String> requestHeaders = getDefaultHeaders(token);
+        for (String key : requestHeaders.keySet()) {
+            getRequest.addHeader(key, requestHeaders.get(key));
+        }
+        HttpResponse response = httpClient.execute(getRequest);
+        if (response.getStatusLine().getStatusCode() != 200) {
+            throw new IOException("Look up for project failed with Status Code: " +
+                    response.getStatusLine().getStatusCode() + " and Status Reason: " +
+                    response.getStatusLine().getReasonPhrase());
+        }
+        JSONObject jsonObject = getJsonFromEntity(response.getEntity());
+        System.out.println("JSON Object: " + jsonObject);
+        JSONArray jsonArray = jsonObject.getJSONArray("data");
+        if (jsonArray.length() == 0) {
+            throw new IOException("No projects found for the API Key: " + apiKey);
+            // throw error
+        }
+        return jsonArray.getJSONObject(0).get("id").toString();
+    }
+
+    private JSONObject getJsonFromEntity(HttpEntity httpEntity) throws IOException {
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(httpEntity.getContent()));
+        StringBuilder sb = new StringBuilder();
+        String line;
+        while ((line = bufferedReader.readLine()) != null) {
+            sb.append(line + '\n');
+        }
+        bufferedReader.close();
+        return new JSONObject(sb.toString());
```

Comment:
[`EntityUtils.toString`](https://hc.apache.org/httpcomponents-core-ga/httpcore/apidocs/org/apache/http/util/EntityUtils.html#toString(org.apache.http.HttpEntity)) will be simpler